core/index/src/main/java/org/locationtech/geowave/core/index/VarintUtils.java

```diff
@@ -284,4 +272,41 @@ public static void writeTime(final long time, final ByteBuffer buffer) {
   public static long readTime(final ByteBuffer buffer) {
     return VarintUtils.readSignedLong(buffer) + TIME_EPOCH;
   }
+
+  /**
+   * Encode a BigDecimal as a byte[]. The structure of the byte[] is opaque, so to deserialize, use
+   * {@link #readBigDecimal(ByteBuffer)}
+   *
+   * @param num The number to serialize as a {@link ByteBuffer}
+   * @return a byte array that represents the given BigDecimal.
+   */
+  public static byte[] writeBigDecimal(final BigDecimal num) {
```

Comment:
Good idea, I just lifted the code I removed from the math-agg and used that instead of this stuff. However, I changed it slightly so it was a better ByteBuffer citizen (storing the array length, instead of using all of `.remaining()`